source/auth/tests/connection-string.js

```diff
@@ -132,6 +150,12 @@ describe("connection string", function() {
 
 			assert.equal(url.credential.mechanism, "PLAIN");
 		});
+
+		it("should throw an exception when an authMechanism is specified with no username", function() {
+			assert.throws(function() {
+				parse("mongodb://localhost/foo/?authMechanism=PLAIN");
```

Comment:
I don't think this extra slash after the dbname is valid and would cause other type of exception